src/main/java/org/advent/daySeven/Hand.java

```diff
@@ -0,0 +1,174 @@
+                listOfCards.add(character);
+            }
+        }
+        return listOfCards;
+    }
+
+    private Type includeJokers(Type type) {
+
+        if (type == Type.TWO_PAIR
+                && anyEqualsJ(relevantCards)) {
+            return Type.FOUR_OF_A_KIND;
+        }
+
+        if (type == Type.TWO_PAIR
+                && anyEqualsJ(restOfHand)) {
+            return Type.FULL_HOUSE;
+        }
+
+        if (type == Type.FULL_HOUSE
+                && anyEqualsJ(relevantCards)) {
+            return Type.FIVE_OF_A_KIND;
+        }
+
+        if (type == Type.FOUR_OF_A_KIND
+                && (anyEqualsJ(restOfHand) || allEqualsJ(relevantCards))) {
+            return Type.FIVE_OF_A_KIND;
+        }
+
+        if (type == Type.THREE_OF_A_KIND
+                && (allEqualsJ(relevantCards) || anyEqualsJ(restOfHand))) {
+            return Type.FOUR_OF_A_KIND;
+        }
+
+        if (type == Type.ONE_PAIR
+                && (allEqualsJ(relevantCards) || anyEqualsJ(restOfHand))) {
+            return Type.THREE_OF_A_KIND;
+        }
+
+        if (type == Type.HIGH_CARD
+                && anyEqualsJ(restOfHand)) {
+            return Type.ONE_PAIR;
+        }
+
+        return type;
+    }
+
+    private boolean anyEqualsJ(List<Character> chars) {
+        return chars.stream().anyMatch(ch -> ch.equals('J'));
+    }
+
+    private boolean allEqualsJ(List<Character> chars) {
+        return chars.stream().allMatch(ch -> ch.equals('J'));
+    }
+
+    private Type typeOfHand(boolean isPartTwo) {
+
+        Type type = null;
+        if (!allCardsMatch() && isListSize(4)) {
+            type = Type.TWO_PAIR;
+        }
+
+        if (!allCardsMatch() && isListSize(5)) {
+            type = Type.FULL_HOUSE;
+        }
+
+        if (allCardsMatch() && isListSize(5)) {
+            type = Type.FIVE_OF_A_KIND;
+        }
+
+        if (allCardsMatch() && isListSize(4)) {
+            type = Type.FOUR_OF_A_KIND;
+        }
+
+        if (allCardsMatch() && isListSize(3)) {
+            type = Type.THREE_OF_A_KIND;
+        }
+
+        if (allCardsMatch() && isListSize(2)) {
+            type = Type.ONE_PAIR;
+        }
+
+        if (relevantCards.isEmpty()) {
+            type = Type.HIGH_CARD;
+        }
+
+        if (isPartTwo) {
+            type = includeJokers(type);
+        }
+
+        return type;
+    }
+
+    private boolean isListSize(int size) {
+        return relevantCards.size() == size;
+    }
+
+    private boolean allCardsMatch() {
+        return relevantCards.stream().allMatch(ch -> ch.equals(relevantCards.get(0)));
+    }
+
+    private String handOfString(String string) {
+        return string.substring(0, string.indexOf(" "));
+    }
+
+    private List<Character> findNonRelevantCardsInHand() {
+        List<Character> listOfNonRelevantCards = new ArrayList<>(handString.chars().mapToObj(ch -> (char) ch).toList());
+        List<Character> listOfRelevantCards = listOfRelevantCards();
```

Comment:
you do work in the constructor, (which is something you shouldnt do), to assign variables with computed values, and then you are not using those variables either